diff --git a/Dockerfile-release b/Dockerfile-release
index 736445f..215cba7 100644
--- a/Dockerfile-release
+++ b/Dockerfile-release
@@ -1,16 +1,10 @@
-FROM alpine:latest
+FROM clefos:latest
 
 ADD etcd /usr/local/bin/
 ADD etcdctl /usr/local/bin/
 RUN mkdir -p /var/etcd/
 RUN mkdir -p /var/lib/etcd/
 
-# Alpine Linux doesn't use pam, which means that there is no /etc/nsswitch.conf,
-# but Golang relies on /etc/nsswitch.conf to check the order of DNS resolving
-# (see https://github.com/golang/go/commit/9dee7771f561cf6aee081c0af6658cc81fac3918)
-# To fix this we just create /etc/nsswitch.conf and add the following line:
-RUN echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf
-
 EXPOSE 2379 2380
 
 # Define default command.
diff --git a/etcdmain/etcd.go b/etcdmain/etcd.go
index a70b250..6250d79 100644
--- a/etcdmain/etcd.go
+++ b/etcdmain/etcd.go
@@ -404,7 +404,7 @@ func setupLogging(cfg *config) {
 
 func checkSupportArch() {
 	// TODO qualify arm64
-	if runtime.GOARCH == "amd64" || runtime.GOARCH == "ppc64le" {
+	if runtime.GOARCH == "amd64" || runtime.GOARCH == "ppc64le" || runtime.GOARCH == "s390x" {
 		return
 	}
 	if env, ok := os.LookupEnv("ETCD_UNSUPPORTED_ARCH"); ok && env == runtime.GOARCH {
