diff --git a/Dockerfile b/Dockerfile
index 3764fbb..a752b1b 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -2,7 +2,7 @@ FROM centos:7
 MAINTAINER Marek Goldmann <mgoldman@redhat.com>
 
 # Install packages necessary to run EAP
-RUN yum update -y && yum -y install xmlstarlet saxon augeas bsdtar unzip && yum clean all
+RUN yum --disablerepo=epel update -y && yum -y --disablerepo=epel install xmlstarlet saxon augeas bsdtar unzip && yum clean all
 
 # Create a user and group used to launch processes
 # The user ID 1000 is the default for the first "regular" user on Fedora/RHEL,
